ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -417,7 +423,62 @@ private AdminResponse handleBlobStoreControlRequest(DataInputStream requestStrea
    */
   private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, AdminRequest adminRequest)
       throws StoreException, IOException {
-    return null;
+    boolean hostHealthy = true; //used to determine if the host is ever unhealthy
+
+    //Finds all replicas of every partition that have this server's hostName
+    List<PartitionId> partitionsWithHostName = new ArrayList<>();
+    StorageManager storageManager = (StorageManager)this.storeManager;
```

Comment:
let's use an if statement to determine if the `storeManager` is a `storageManager`. If not, we can just return "BAD".

Reply:
Done